lib/src/utilities/simcompare.dart

```diff
@@ -107,14 +108,15 @@ abstract class SimCompare {
             for (final signalName in vector.expectedOutputValues.keys) {
               final value = vector.expectedOutputValues[signalName];
               final o = module.output(signalName);
+
               final errorReason =
                   'For vector #${vectors.indexOf(vector)} $vector,'
                   ' expected $o to be $value, but it was ${o.value}.';
               if (value is int) {
                 if (!o.value.isValid) {
                   // invalid value causes exception without helpful message,
                   // so throw it
-                  throw Exception(errorReason);
+                  throw MismatchOutputValueException(vectors, vector, o, value);
                 }
```

Comment:
We could do `expect(o.value.isValid, isTrue)` instead of throwing an exception here